stub-idp/src/main/java/uk/gov/ida/stub/idp/resources/idp/LoginPageResource.java

```diff
@@ -228,6 +249,65 @@ public Response postAuthnPending(
         return samlResponseRedirectViewFactory.sendSamlMessage(pendingResponse);
     }
 
+
+    private boolean sessionContainsUser(Optional<IdpSession> session) {
+        return session.isPresent() && session.get().getIdpUser().isPresent();
+    }
+
+    private boolean sessionHasIdaAuthnRequestFromHub(Optional<IdpSession> session) {
+        return session.isPresent() && session.get().getIdaAuthnRequestFromHub() != null;
+    }
+
+    private Response createSessionAndShowLoginForm(Idp idp, Optional<ErrorMessageType> errorMessage){
+
+        final IdpSession idpSession = new IdpSession(
```

Comment:
I'd be tempted to have an overloaded constructor that takes just two parameters (`sessionId` and `singleIdpJourneyId` rather than passing all the `null` values.